src/containers/User/Login.tsx

```diff
@@ -1,68 +1,142 @@
-import React, { useState } from 'react'
+import React, { useState, useEffect } from 'react'
 import { useRouter } from '../../utils/hooks/useRouter'
 import { useUserState } from '../../contexts/UserState'
 import { Link } from 'react-router-dom'
-import { Form, Button, Container, Grid, Segment, Header } from 'semantic-ui-react'
+import { Form, Button, Container, Grid, Segment, Header, Dropdown } from 'semantic-ui-react'
 import config from '../../config.json'
 import isLoggedIn from '../../utils/helpers/loginCheck'
 import strings from '../../utils/constants'
-import setUserInfo from '../../utils/helpers/fetchUserInfo'
+import messages from '../../utils/messages'
+import { attemptLogin, attemptLoginOrg } from '../../utils/helpers/attemptLogin'
+import { LoginPayload, OrganisationSimple } from '../../utils/types'
 
 const Login: React.FC = () => {
   const [username, setUsername] = useState('')
   const [password, setPassword] = useState('')
   const [isError, setIsError] = useState(false)
+  const [networkError, setNetworkError] = useState('')
+  const [loginPayload, setLoginPayload] = useState<LoginPayload>()
+  const [selectedOrgIndex, setSelectedOrgIndex] = useState(0)
   const { push, history } = useRouter()
-  const { login } = useUserState()
+  const { onLogin } = useUserState()
 
-  const handleSubmit = async (event: any) => {
-    event.preventDefault()
-    const loginResult = await attemptLogin(username, password)
-    if (!loginResult.success) setIsError(true)
-    else {
-      setIsError(false)
-      login(loginResult.JWT)
-      if (history.location?.state?.from) push(history.location.state.from)
-      else push('/')
+  // useEffect ensures isLoggedIn only runs on first mount, not re-renders
+  useEffect(() => {
+    if (isLoggedIn()) push('/')
+  }, [])
+
+  const onLoginSuccess = (loginResult: LoginPayload) => {
+    setIsError(false)
+    setLoginPayload({
+      ...loginResult,
+    })
+  }
+
+  const handleSubmit = async () => {
+    setNetworkError('')
+    if (!loginPayload) {
+      // User login
+      await attemptLogin({
+        username,
+        password,
+        onLoginSuccess,
+        onLoginFailure: () => setIsError(true),
+      }).catch((error) => {
+        setNetworkError(error.message)
+      })
+    } else {
+      // Organisation login
+      const { orgId } = loginPayload?.orgList?.[selectedOrgIndex] as OrganisationSimple
+      attemptLoginOrg({
```

Comment:
```suggestion
      await attemptLoginOrg({
```

I guess this one should also await?

Reply:
> I guess this one should also await?

Actually, neither need `await`, since there's nothing afterwards which relies on waiting sequentially for them. Have removed from both (and added `.catch` for earlier one).